test/test_auth_endpoint.py

```diff
@@ -75,31 +86,33 @@ def test_oidc_authorization_code_non_irods_user():
     res = s.post(extracted_url, data={'username': 'non_irods_user', 'password': 'bad', 'credentialId': ''})
     assert res.status_code == requests.codes.bad_request
 
-
 @pytest.mark.parametrize("username, password, expected_result", [('rods', 'rods', requests.codes.ok),
                                                                  ('', '', requests.codes.unauthorized),
                                                                  ('not', 'valid', requests.codes.unauthorized),
                                                                  ('a'*200, 'b'*200, requests.codes.unauthorized)])
-def test_post_basic_login(username, password, expected_result):
-    res = requests.post(AUTH_URL, auth=(username, password))
+def test_post_basic_login(username, password, expected_result, auth_url):
+    res = requests.post(auth_url, auth=(username, password))
 
     # Got a good code, assume we passed...
     assert res.status_code == expected_result
 
+@pytest.mark.parametrize("state, code", [('placeholder', 'bad_code'),
+                                         ('', 'bad_code'),
+                                         ('placeholder', ''),
+                                         ('a'*200, 'b'*200)])
+def test_get_oidc_errors(state, code, auth_url):
+    res = requests.get(auth_url, params={'state': state, 'code': code})
 
-@pytest.mark.parametrize("state, code, expected_result", [('placeholder', 'bad_code', requests.codes.bad_request),
-                                                          ('', 'bad_code', requests.codes.bad_request),
-                                                          ('placeholder', '', requests.codes.bad_request),
-                                                          ('a'*200, 'b'*200, requests.codes.bad_request)])
-def test_get_oidc_errors(state, code, expected_result):
-    res = requests.get(AUTH_URL, params={'state': state, 'code': code})
-
-    assert res.status_code == expected_result
+    if test_config['openid_connect']['mode'] == 'client':
```

Comment:
Probably should check for existence of properties?

Reply:
Resolved with schema.